packages/react-utilities/src/hooks/useOnClickOutside.ts

```diff
@@ -0,0 +1,26 @@
+import * as React from 'react';
+import { useEventCallback } from './useEventCallback';
+
+export const useOnClickOutside = (options: {
+  element?: Node | Window | Document;
+  refs: React.MutableRefObject<HTMLElement>[];
```

Comment:
```suggestion
  refs: React.MutableRefObject<HTMLElement | undefined | null>[];
```

_nit:_ We probably should relax typings a bit to simplify further usages